src/main/java/org/spongepowered/api/event/lifecycle/RegisterChannelEvent.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.lifecycle;
+
+import org.spongepowered.api.ResourceKey;
+import org.spongepowered.api.network.channel.Channel;
+import org.spongepowered.api.network.channel.ChannelRegistry;
+import org.spongepowered.api.registry.DuplicateRegistrationException;
+
+/**
+ * Lifecycle event to indicate when network channels should be created and registered.
+ */
+public interface RegisterChannelEvent extends LifecycleEvent {
+
+    /**
+     * Gets the {@link ChannelRegistry}.
+     *
+     * @return The channel registry
+     */
+    ChannelRegistry getRegistry();
```

Comment:
For what purpose would you expose the registry at this time?

The only goal of this event is registering a channel. If they want to query for another channel, best do that in LoadedGameEvent.